Cython/Utility/Profile.c

```diff
@@ -33,8 +33,229 @@
+      memset($monitoring_states_cname, 0, sizeof($monitoring_states_cname));                 \
+      if (nogil) {                                                                           \
+          if (CYTHON_TRACE_NOGIL) {                                                          \
+              PyGILState_STATE state = PyGILState_Ensure();                                  \
+              if (unlikely(!$frame_code_cname)) $frame_code_cname = (PyObject*) __Pyx_createFrameCodeObject(funcname, srcfile, __Pyx_offset_to_line(offset)); \
+              if (unlikely(!$frame_code_cname)) goto_error;                                  \
+              ret = __Pyx__TraceStartFunc($monitoring_states_cname, $frame_code_cname, offset, skip_event); \
+              PyGILState_Release(state);                                                     \
+          }                                                                                  \
+      } else {                                                                               \
+          if (unlikely(!$frame_code_cname)) $frame_code_cname = (PyObject*) __Pyx_createFrameCodeObject(funcname, srcfile, __Pyx_offset_to_line(offset)); \
+          if (unlikely(!$frame_code_cname)) goto_error;                                      \
+          ret = __Pyx__TraceStartFunc($monitoring_states_cname, $frame_code_cname, offset, skip_event); \
+      }                                                                                      \
+      if (unlikely(ret == -1)) goto_error;                                                   \
+  }
+
+  // 'nogil' is obviously unused
+  #define __Pyx_TraceStartGen(funcname, srcfile, offset, nogil, skip_event, goto_error) \
+  if ((0) /* !__Pyx_IsTracing(__Pyx_Monitoring_PY_START) */); else {                         \
+      int ret = __Pyx__TraceStartGen($monitoring_states_cname, &$monitoring_version_cname, $frame_code_cname, offset); \
+      if (unlikely(ret == -1)) goto_error;                                                   \
+  }
+
+  #define __Pyx_TraceResumeGen(funcname, srcfile, firstlineno, offset, goto_error) \
+  if ((0) /* !__Pyx_IsTracing(__Pyx_Monitoring_PY_RESUME) */); else {                        \
+      int ret = __Pyx__TraceResumeGen($monitoring_states_cname, &$monitoring_version_cname, $frame_code_cname, offset); \
+      if (unlikely(ret == -1)) goto_error;                                                   \
+  }
+
+  #define __Pyx_TraceYield(result, offset, goto_error) \
+  if (!__Pyx_IsTracing(__Pyx_Monitoring_PY_YIELD)); else {                                   \
+      int ret = PyMonitoring_FirePyYieldEvent(&$monitoring_states_cname[__Pyx_Monitoring_PY_RETURN], $frame_code_cname, offset, result); \
+      PyMonitoring_ExitScope();                                                              \
+      if (unlikely(ret == -1)) goto_error;                                                   \
+  }
+
+//   #if PY_VERSION_HEX >= 0x030d00b2
```

Comment:
This code is forgotten and should be removed?

Reply:
It's something that is still missing from (although not required for) the generator implementation but I'm not sure yet how and when to report the event. I'll leave the code in as a reminder that it's an open issue.